File: data/derived_cot/rq1_traces/translation/qwen/java_to_cpp/code_output/ComplexCalculator.cpp
#include <cmath>
#include <sstream>
#include <stdexcept>
#include <string>

class ComplexCalculator {
public:
    class ComplexNumber {
    private:
        double real;
        double imaginary;

    public:
        ComplexNumber(double real, double imaginary) : real(real), imaginary(imaginary) {}

        double getReal() const { return real; }
        double getImaginary() const { return imaginary; }

        bool operator==(const ComplexNumber& other) const {
            // Handle NaN cases as in Java's Double.compare
            if (std::isnan(real) && std::isnan(other.real) && 
                std::isnan(imaginary) && std::isnan(other.imaginary)) {
                return true;
            }
            if (std::isnan(real) || std::isnan(other.real) ||
                std::isnan(imaginary) || std::isnan(other.imaginary)) {
                return false;
            }
            return (real == other.real) && (imaginary == other.imaginary);
        }

        std::string toString() const {
            std::ostringstream oss;
            oss << real;
            if (imaginary >= 0) {
                oss << "+";
            }
            oss << imaginary << "j";
            return oss.str();
        }
    };

    static ComplexNumber add(const ComplexNumber& c1, const ComplexNumber& c2) {
        double real = c1.getReal() + c2.getReal();
        double imaginary = c1.getImaginary() + c2.getImaginary();
        return ComplexNumber(real, imaginary);
    }

    static ComplexNumber subtract(const ComplexNumber& c1, const ComplexNumber& c2) {
        double real = c1.getReal() - c2.getReal();
        double imaginary = c1.getImaginary() - c2.getImaginary();
        return ComplexNumber(real, imaginary);
    }

    static ComplexNumber multiply(const ComplexNumber& c1, const ComplexNumber& c2) {
        double real = c1.getReal() * c2.getReal() - c1.getImaginary() * c2.getImaginary();
        double imaginary = c1.getReal() * c2.getImaginary() + c1.getImaginary() * c2.getReal();
        return ComplexNumber(real, imaginary);
    }

    static ComplexNumber divide(const ComplexNumber& c1, const ComplexNumber& c2) {
        double denominator = c2.getReal() * c2.getReal() + c2.getImaginary() * c2.getImaginary();
        if (denominator == 0) {
            throw std::runtime_error("Division by zero");
        }
        double real = (c1.getReal() * c2.getReal() + c1.getImaginary() * c2.getImaginary()) / denominator;
        double imaginary = (c1.getImaginary() * c2.getReal() - c1.getReal() * c2.getImaginary()) / denominator;
        return ComplexNumber(real, imaginary);
    }
};

File: data/derived_cot/rq1_traces/translation/qwen/java_to_py/code_output/ComplexCalculator.py
class ComplexCalculator:

    class ComplexNumber:
        def __init__(self, real, imaginary):
            self.real = real
            self.imaginary = imaginary

        def getReal(self):
            return self.real

        def getImaginary(self):
            return self.imaginary

        def __eq__(self, other):
            if isinstance(other, ComplexCalculator.ComplexNumber):
                return self.real == other.real and self.imaginary == other.imaginary
            return False

        def __str__(self):
            if self.imaginary == 0:
                return f"{self.real}+0j"
            elif self.real == 0:
                if self.imaginary >= 0:
                    return f"{self.imaginary}j"
                else:
                    return f"-{abs(self.imaginary)}j"
            else:
                if self.imaginary >= 0:
                    return f"{self.real}+{self.imaginary}j"
                else:
                    return f"{self.real}{self.imaginary}j"

    @staticmethod
    def add(c1, c2):
        real = c1.getReal() + c2.getReal()
        imaginary = c1.getImaginary() + c2.getImaginary()
        return ComplexCalculator.ComplexNumber(real, imaginary)

    @staticmethod
    def subtract(c1, c2):
        real = c1.getReal() - c2.getReal()
        imaginary = c1.getImaginary() - c2.getImaginary()
        return ComplexCalculator.ComplexNumber(real, imaginary)

    @staticmethod
    def multiply(c1, c2):
        real = c1.getReal() * c2.getReal() - c1.getImaginary() * c2.getImaginary()
        imaginary = c1.getReal() * c2.getImaginary() + c1.getImaginary() * c2.getReal()
        return ComplexCalculator.ComplexNumber(real, imaginary)

    @staticmethod
    def divide(c1, c2):
        denominator = c2.getReal() * c2.getReal() + c2.getImaginary() * c2.getImaginary()
        real = (c1.getReal() * c2.getReal() + c1.getImaginary() * c2.getImaginary()) / denominator
        imaginary = (c1.getImaginary() * c2.getReal() - c1.getReal() * c2.getImaginary()) / denominator
        return ComplexCalculator.ComplexNumber(real, imaginary)